src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -167,6 +167,33 @@ public SnapshotService(
     this.snapshotBuilderSettingsDao = snapshotBuilderSettingsDao;
   }
 
+  public String getSnapshotName(SnapshotRequestModel model) {
+    SnapshotRequestContentsModel contentsModel = model.getContents().get(0);
+    if (contentsModel.getMode() == SnapshotRequestContentsModel.ModeEnum.BYREQUESTID) {
+      SnapshotAccessRequestResponse snapshotAccessRequestResponse =
+          snapshotRequestDao.getById(contentsModel.getRequestIdSpec().getSnapshotRequestId());
+      String dashesAndSpacesRegex = "[- ]+";
+      String nonAlphaNumericRegex = "\\W";
+
+      // We take the first 474 characters of the name, because the UUID is 36 characters long, so
+      // the name is 474 characters + the underscore + 36 characters for the UUID, totalling to a
+      // max of 511.
+      String generatedName =
+          String.format(
+              "%s_%s",
+              snapshotAccessRequestResponse.getSnapshotName().substring(0, 473),
```

Comment:
I think this will throw, unless the name is longer than 473 characters. Using`StringUtils.truncate()` would work.

Also I would write this using math
```
  static final int SNAPSHOT_NAME_MAX = 511;

...
  String id = accessRequest.getId().toString();
  int maxLength = SNAPSHOT_NAME_MAX - (id.length + 1);
  ...
```

Also I think you want to truncate after cleaning as that could make the snapshot name shorter.